streams/src/test/java/org/apache/kafka/streams/integration/ConsistencyVectorIntegrationTest.java

```diff
@@ -0,0 +1,236 @@
+    private static final String TABLE_NAME = "source-table";
+
+    public final EmbeddedKafkaCluster cluster = new EmbeddedKafkaCluster(NUM_BROKERS);
+
+    @Rule
+    public TestName testName = new TestName();
+
+    private final List<KafkaStreams> streamsToCleanup = new ArrayList<>();
+    private final MockTime mockTime = cluster.time;
+
+    @Before
+    public void before() throws InterruptedException, IOException {
+        cluster.start();
+        cluster.createTopic(INPUT_TOPIC_NAME, 2, 1);
+    }
+
+    @After
+    public void after() {
+        for (final KafkaStreams kafkaStreams : streamsToCleanup) {
+            kafkaStreams.close();
+        }
+        cluster.stop();
+    }
+
+    @Test
+    public void shouldHaveSamePositionBoundActiveAndStandBy() throws Exception {
+        final int batch1NumMessages = 100;
+        final int key = 1;
+        final Semaphore semaphore = new Semaphore(0);
+
+        final StreamsBuilder builder = new StreamsBuilder();
+        Objects.requireNonNull(TABLE_NAME, "name cannot be null");
+        final TestingRocksDbKeyValueBytesStoreSupplier supplier =
+                new TestingRocksDbKeyValueBytesStoreSupplier(TABLE_NAME);
+
+        builder.table(INPUT_TOPIC_NAME, Consumed.with(Serdes.Integer(), Serdes.Integer()),
+                        Materialized.<Integer, Integer>as(supplier).withCachingDisabled())
+                .toStream()
+                .peek((k, v) -> semaphore.release());
+
+        final KafkaStreams kafkaStreams1 = createKafkaStreams(builder, streamsConfiguration());
+        final KafkaStreams kafkaStreams2 = createKafkaStreams(builder, streamsConfiguration());
+        final List<KafkaStreams> kafkaStreamsList = Arrays.asList(kafkaStreams1, kafkaStreams2);
+
+        startApplicationAndWaitUntilRunning(kafkaStreamsList, Duration.ofSeconds(60));
+
+        produceValueRange(key, 0, batch1NumMessages);
+
+        // Assert that all messages in the first batch were processed in a timely manner
+        assertThat(semaphore.tryAcquire(batch1NumMessages, 60, TimeUnit.SECONDS), is(equalTo(true)));
+
+        final QueryableStoreType<ReadOnlyKeyValueStore<Integer, Integer>> queryableStoreType = keyValueStore();
+
+        // Assert that both active and standby have the same position bound
+        TestUtils.waitForCondition(() -> {
+            final ReadOnlyKeyValueStore<Integer, Integer> store1 = getStore(TABLE_NAME, kafkaStreams1, true, queryableStoreType);
+            return store1.get(key) != null;
+        }, "store1 cannot find results for key");
+        TestUtils.waitForCondition(() -> {
+            final ReadOnlyKeyValueStore<Integer, Integer> store2 = getStore(TABLE_NAME, kafkaStreams2, true, queryableStoreType);
+            return store2.get(key) != null;
+        }, "store2 cannot find results for key");
+
+        final AtomicInteger count = new AtomicInteger();
+        for (final TestingRocksDBStore store : supplier.stores) {
```

Comment:
I had written a comment here but it got lost: There must be a better way to test this: I want to make sure that at the end, there are two stores that have partition 0, one at the active and one at the standby so that I can compare their position.

Reply:
I think there will be a better way to write this test once we have the position tracking implemented end-to-end. For now, I think this test is sufficient.